File: app/src/main/java/com/ikmr/banbara23/yaeyama_liner_checker/common/StatusHelper.java
package com.ikmr.banbara23.yaeyama_liner_checker.common;

import android.support.v4.content.ContextCompat;

import com.ikmr.banbara23.yaeyama_liner_checker.R;
import com.ikmr.banbara23.yaeyama_liner_checker.core.Base;

public class StatusHelper {

    public static int getStatusColor(String code) {
        if (code == null) {
            return ContextCompat.getColor(Base.getContext(), R.color.status_cation);
        }
        switch (code) {
            case "normal":
                return ContextCompat.getColor(Base.getContext(), R.color.status_normal);
            case "cancel":
            case "suspend":
                return ContextCompat.getColor(Base.getContext(), R.color.status_cancel);
            case "cation":
            default:
                return ContextCompat.getColor(Base.getContext(), R.color.status_cation);
        }
    }
}

File: app/src/main/java/com/ikmr/banbara23/yaeyama_liner_checker/common/StatusHelper.kt
package com.ikmr.banbara23.yaeyama_liner_checker.common

import android.graphics.drawable.Drawable
import androidx.core.content.ContextCompat
import com.ikmr.banbara23.yaeyama_liner_checker.R
import com.ikmr.banbara23.yaeyama_liner_checker.core.Base

object StatusHelper {

    @JvmStatic
    fun getStatusColor(code: String?): Drawable? {
        return when (code) {
            "normal" -> ContextCompat.getDrawable(Base.getContext(), R.drawable.shape_rounded_corners_status_nomal)
            "cancel", "suspend" -> ContextCompat.getDrawable(Base.getContext(), R.drawable.shape_rounded_corners_status_cancel)
            "cation" -> ContextCompat.getDrawable(Base.getContext(), R.drawable.shape_rounded_corners_status_cation)
            else -> ContextCompat.getDrawable(Base.getContext(), R.drawable.shape_rounded_corners_status_cancel)
        }
    }
}

